### DISCUSSION TITLE
Debugging and/or console logging

### DISCUSSION THREAD

**Discussion by msurdi** [statelyai/xstate#2460] [Q&A] [answered]
Hi!

I'm just starting to learn XState and so far I find it great. Something I'm stumbling upon and couldn't find anything in the docs and searching around is if there is any way to debug/print every state change and/or event to the console other than `@xstate/inspect`.

Thanks!

**Comment by davidkpiano** [comment, marked as answer]
Glad you're liking it! There is a way: you can **subscribe** to state changes and do whatever you want in the observer:

```js
const service = interpret(someMachine);

service.subscribe(state => {
  console.log(state);
});

service.start();
```
